### PR TITLE
Add dynamic colour change feature using keyboard inputs

### PR DESCRIPTION
**Pull request by DinethDilhara**
### Description
This pull request introduces a feature that allows users to change the sparkle color dynamically during the animation by pressing specific keys:
- `R` for red
- `G` for green
- `B` for blue
- `W` for white (default)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -23,18 +23,18 @@ struct Sparkle {
...
-        let color = Color::new(r, g, b, alpha);
+        // Set the color with varying alpha
+        let mut color = color;
+        color.a = alpha;
+
+        // Draw the circle
         draw_circle(self.position.x, self.position.y, self.radius, color);
     }
-}
+}
```
Comment (bahamas10) [inline src/main.rs]: extra whitespace here

File: src/main.rs [hunk 2]
```diff
@@ -49,6 +49,11 @@ fn conf() -> Conf {
 
 #[macroquad::main(conf)]
 async fn main() {
+
```
Comment (bahamas10) [inline src/main.rs]: extra newline here shouldn't be here

**Review by bahamas10 — COMMENTED** [review]
this is awesome! thank for the PR - i've commented 2 nits i have... i think if you run:

```
cargo fmt
```

it should fix these for you automatically

File: src/main.rs (hunk 1 above)
Comment (DinethDilhara) [reply src/main.rs]: Fixed the nits as suggested. Thanks for the review!

**Comment by bahamas10** [comment]
LGTM! ty for the pr

**Pull request merged**